Mage/src/main/java/mage/abilities/effects/common/continuous/GainAbilityControlledSpellsEffect.java

```diff
@@ -46,22 +46,22 @@ public boolean apply(Game game, Ability source) {
         }
 
         for (Card card : game.getExile().getAllCardsByRange(game, source.getControllerId())) {
-            if (card.isOwnedBy(source.getControllerId()) && filter.match(card, game)) {
+            if (filter.match(card, game) && player.getPlayableObjects(game, Zone.EXILED, true).containsObject(card.getId())) {
```

Comment:
`getPlayable` code:
- related to GUI and AI only
- it's limited for some dynamic use cases and do not show a real playable objects/abilities (example: some restriction/replacement effects for attack/defend/mana usage -- applied after a real use only, not before);
- it's depends on current game cycle result (e.g. on other applied effects);
- so getPlayable usage in the cards is bad workaround and must be removed;

Reply:
Yes, I see, that was not the right direction.